modules/bene_promo_redirect/js/bene-promo-redirect.js

```diff
@@ -0,0 +1,17 @@
+let validReferrer = false;
+if (document.referrer === "") {
+  validReferrer = true;
+}
+else {
+  let ref = new URL(document.referrer);
+  if (ref.hostname !== drupalSettings.benePromoRedirect.destinationDomain && ref.hostname !== window.location.hostname) {
+    validReferrer = true;
+  }
+}
+
+if (validReferrer) {
+  let urlParams = new URLSearchParams(window.location.search);
+  if (urlParams.get('promo_redirect') !== 'no') {
+    window.location.href = drupalSettings.benePromoRedirect.destination;
```

Comment:
How does the "until the user has visited the target" part of the feature get enforced?

Reply:
Line 7: we compare the referrer to the "destination domain". So, if they are returning from whence they came, or coming from inside the site, the validReferrer is not set.

If the remote site isn't cooperative with this referrer logic for whatever reason, then the links on that site will need to be edited to include the `?promo_redirect=no` query.